tests/integration/pod_args/test_podman_compose_pod_args.py

```diff
@@ -0,0 +1,176 @@
+# SPDX-License-Identifier: GPL-2.0
+
+import json
+import os
+import unittest
+
+from tests.integration.test_utils import RunSubprocessMixin
+
+
+def base_path():
+    """Returns the base path for the project"""
+    return os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(__file__))))
+
+
+def test_path():
+    """Returns the path to the tests directory"""
+    return os.path.join(base_path(), "tests/integration")
+
+
+def podman_compose_path():
+    """Returns the path to the podman compose script"""
+    return os.path.join(base_path(), "podman_compose.py")
+
+
+class TestPodmanComposePodArgs(unittest.TestCase, RunSubprocessMixin):
+    def run_pod_args_test(self, config, args, expected):
+        """
+        Helper to run podman up with a docker-compose.yml config, additional
+        (--pod-args) arguments and compare the CreateCommand of the resulting
+        pod with an expected value
+        """
+        pod_name = "pod_" + config
+        command_up = (
+            [
+                "python3",
+                os.path.join(base_path(), "podman_compose.py"),
+                "-f",
+                os.path.join(
+                    base_path(),
+                    "tests",
+                    "integration",
+                    "pod_args",
+                    config,
+                    "docker-compose.yml",
+                ),
+            ]
+            + args
+            + [
+                "up",
+                "--no-start",
+            ]
+        )
+
+        try:
+            self.run_subprocess_assert_returncode(command_up)
+
+            output, _ = self.run_subprocess_assert_returncode([
+                "podman",
+                "pod",
+                "inspect",
+                pod_name,
+            ])
+            pod_info = json.loads(output.decode('utf-8'))
+            # Podman 5.0 changed pod inspect to always output a list.
+            # Check type to support both old and new version.
+            if isinstance(pod_info, list):
+                pod_info = pod_info[0]
```

Comment:
Makes sense to extract the above 5 lines into separate function, e.g. `load_pod_info`.

Reply:
Done